File: 0028.ImplementStrstr/ImplementStrstr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        len1, len2 = len(haystack), len(needle)
        if len1 < len2:
            return -1
        if len2 == 0:
            return 0
        for i in range(len1 - len2 + 1):
            if haystack[i:i + len2] == needle:
                return i
        return -1

File: 0028.ImplementStrstr/ImplementStrstr.c


int strStr(char *haystack, char *needle)
{
        int hlen = strlen(haystack);
        int nlen = strlen(needle);
        if (hlen < nlen)
                return -1;
        if (!nlen)
                return 0;

        int i, j, k;
        for (i = 0; i <= hlen - nlen; i++)
        {
                k = 1;
                for (j = 0; j < nlen; j++)
                {
                        if (haystack[i + j] != needle[j])
                        {
                                k = 0;
                                break;
                        }
                }
                if (k)
                        return i;
        }

        return -1;
}
